homeassistant/components/smart_meter_b_route/config_flow.py

```diff
@@ -0,0 +1,131 @@
+    with Momonga(dev=device, rbid=id, pwd=password):
+        pass
+
+
+def _human_readable_device_name(port: UsbServiceInfo | ListPortInfo) -> str:
+    return human_readable_device_name(
+        port.device,
+        port.serial_number,
+        port.manufacturer,
+        port.description,
+        str(port.vid) if port.vid else None,
+        str(port.pid) if port.pid else None,
+    )
+
+
+def _generate_unique_id(port: UsbServiceInfo | ListPortInfo) -> str:
+    """Generate unique id from usb attributes."""
+    return f"{port.vid}:{port.pid}_{port.serial_number}_{port.manufacturer}_{port.description}"
+
+
+class BRouteConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Smart Meter B Route."""
+
+    VERSION = 1
+
+    device: UsbServiceInfo | None = None
+
+    @callback
+    def _get_discovered_device_id_and_name(
+        self, device_options: dict[str, ListPortInfo]
+    ) -> tuple[str | None, str | None]:
+        discovered_device_id = (
+            get_serial_by_id(self.device.device) if self.device else None
+        )
+        discovered_device = (
+            device_options.get(discovered_device_id) if discovered_device_id else None
+        )
+        discovered_device_name = (
+            _human_readable_device_name(discovered_device)
+            if discovered_device
+            else None
+        )
+        return discovered_device_id, discovered_device_name
+
+    async def _get_usb_devices(self) -> dict[str, ListPortInfo]:
+        """Return a list of available USB devices."""
+        devices = await self.hass.async_add_executor_job(comports)
+        return {get_serial_by_id(port.device): port for port in devices}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        device_options = await self._get_usb_devices()
+        if user_input is not None:
+            try:
+                await self.hass.async_add_executor_job(
+                    _validate_input,
+                    user_input[CONF_DEVICE],
+                    user_input[CONF_ID],
+                    user_input[CONF_PASSWORD],
+                )
+            except MomongaSkScanFailure:
+                errors["base"] = "cannot_connect"
+            except MomongaSkJoinFailure:
+                errors["base"] = "invalid_auth"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                device_id = get_serial_by_id(user_input[CONF_DEVICE])
+                device = device_options[device_id]
+                if not self.context.get("unique_id"):
+                    await self.async_set_unique_id(
+                        _generate_unique_id(device), raise_on_progress=False
+                    )
+                    self._abort_if_unique_id_configured()
```

Comment:
in theory the unique id wouldnt be set right?

Reply:
Sorry, not sure I understand the comment :D

It is set just above

Reply:
Why the 
```
                if not self.context.get("unique_id"):
```

Reply:
Removed